File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int n=nums.length;
        int currentSum=0,maxSum=-999999;
        for(int i=0;i<n;i++)
        {
            currentSum+=nums[i];
            maxSum=max(currentSum,maxSum);
            if(currentSum<0)
            {
                currentSum=0;
            }
        }
        return maxSum;
    }
    public int max(int a,int b)
    {
        if(a>b)
            return a;
        return b;
    }
}

File: 53-maximum-subarray/maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        currentSum=0
        maxSum=-float('inf')
        for val in nums:
            currentSum+=val
            maxSum=max(currentSum,maxSum)
            if(currentSum<0):
                currentSum=0
        return maxSum